libass/ass_filesystem.c

```diff
@@ -50,6 +50,43 @@ static bool alloc_path(ASS_Dir *dir, size_t size)
 }
 
 
+#if HAVE_FSTAT || _WIN32
+// Microsoft docs require types.h to be included before stat.h
+// IN POSIX just stat.h suffices
+#if _WIN32
+#include <sys/types.h>
+#endif
+#include <sys/stat.h>
```

Comment:
It is true that POSIX seems to require `sys/stat.h` alone to define everything necessary, but my Linux man page does list `sys/types.h` among the necessary includes and even adds `unistd.h`. I assume that there must be some older/less POSIX-compliant systems (besides Windows) out there that require `sys/types.h` before `sys/stat.h`. Why not just include both unconditionally?

Reply:
seems to have been a (non)change in SUSv3/POSIX issue 6 (2004); from the change history:
> The requirement to include [<sys/types.h>](https://pubs.opengroup.org/onlinepubs/009695399/basedefs/sys/types.h.html) has been removed. Although [<sys/types.h>](https://pubs.opengroup.org/onlinepubs/009695399/basedefs/sys/types.h.html) was required for conforming implementations of previous POSIX specifications, it was not required for UNIX applications.

Now including both uncoditionally